### PR TITLE
Add ToJSON and FromJSON instances for Address

### PR DESCRIPTION
**Pull request by MarcFontaine**

### CODE REVIEW

**Comment by MarcFontaine** [comment]
These are the reusable bits from https://github.com/input-output-hk/cardano-node/pull/1888.
The other parts of the functionality from #1888 have landed in master already.

File: cardano-api/src/Cardano/Api/Address.hs [hunk 1]
```diff
@@ -263,6 +263,25 @@ instance SerialiseAddress (Address ShelleyAddr) where
...
+    parseJSON = Aeson.withText "Address" $ \txt ->
+      maybe
+        (fail "Invalid Byron address.")
+        pure
+        (deserialiseAddress AsByronAddress txt)
+
+instance FromJSON (Address ShelleyAddr) where
+    parseJSON = Aeson.withText "Address" $ \txt ->
+      maybe
+        (fail "Invalid Shelley address.")
```
Comment (Jimbo4350) [inline cardano-api/src/Cardano/Api/Address.hs:282]: I would update the error messages to "Cardano.Api.Address.FromJSON: Invalid $era address"

**Review by Jimbo4350 — APPROVED** [review]
LGTM! One minor change

File: cardano-api/src/Cardano/Api/Address.hs (hunk 1 above)
Comment (MarcFontaine) [reply cardano-api/src/Cardano/Api/Address.hs:282]: OK done that

**Pull request merged**